classes/02class/exercises/c04-iam01/mendoncaj/cli_commands.txt

```diff
@@ -0,0 +1,102 @@
+> aws iam create-user --user-name user_readonly_S3
+
+**output**:
+```
+{
+    "User": {
+        "Path": "/",
+        "UserName": "user_readonly_S3",
+        "UserId": "AIDAR6HBDI4EW3NAA7CQ3",
+        "Arn": "arn:aws:iam::133616060169:user/user_readonly_S3",
+        "CreateDate": "2020-07-03T02:58:33Z"
+    }
+}
+```
+
+> aws iam create-policy \
+--policy-name S3readOnly \
+--policy-document file://~/aws-policies/user_readonly_S3-policy.json
+
+```
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:Get*",
+        "s3:List*"
+      ],
+      "Effect": "Allow",
+      "Resource": [
+        "arn:aws:s3:::*",
+        "arn:aws:s3:::*/*"
+      ]
+      "Principal": {
+        "AWS": [
+          "user_readonly_S3"
+        ]
+      }
```

Comment:
This line should not be here.
**Principal** element must be used for [IAM roles or resource-based policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_principal.html).

For this exercise, we are creating an IAM identity based policy which will be **attached** to the user _user_readonly_S3_.